Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs

```diff
@@ -86,20 +83,25 @@ public static bool InheritsFromOrImplements(this TypeReference typeRef, TypeRefe
 		{
 			var arrayInterfaces = new[]
 			{
-				"System.IEnumerable",
+				"System.Collections.IEnumerable",
 				"System.Collections.IList",
 				"System.Collections.Collection"
 			};
 
 			var arrayGenericInterfaces = new[]
 			{
-				"System.IEnumerable`1",
+				"System.Collections.IEnumerable`1",
 				"System.Collections.Generic.IList`1",
 				"System.Collections.Generic.IReadOnlyCollection<T>",
 				"System.Collections.Generic.IReadOnlyList<T>",
 				"System.Collections.Generic.Collection<T>"
 			};
 
+			if (typeRef.IsArray && baseClass.IsArray) {
```

Comment:
if both are arrays, say object[] and string[], resolve the type and apply the test on object and string